core/object/class_db.h

```diff
@@ -249,6 +269,9 @@ class ClassDB {
 			argptrs[i] = &args[i];
 		}
 		MethodBind *bind = create_method_bind(p_method);
+		if (std::is_same<typename member_function_traits<M>::return_type, Object *>::value) {
```

Comment:
I believe this (and other similar conditions) can be `if constexpr`. Not sure if the compiler optimizes this without it.

Reply:
It's likely, but why take the risk? Pushed with `constexpr` in place.